### PR TITLE
Add logging to FlakyTestRetryAnalyzer

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-testng-services/src/main/java/io/prestosql/testng/services/FlakyTestRetryAnalyzer.java [hunk 1]
```diff
@@ -49,22 +49,30 @@
...
+            log.info("not retrying; @Flaky annotation not present");
             return false;
         }
         if (result.getThrowable() == null) {
+            log.info("not retrying; throwable not present in result");
             return false;
         }
         String stackTrace = getStackTraceAsString(result.getThrowable());
         if (!Pattern.compile(annotation.match()).matcher(stackTrace).find()) {
+            log.info("not retrying; stacktrace '" + stackTrace + "' does not match pattern '" + annotation.match() + "'");
```
Comment (findepi) [inline presto-testng-services/src/main/java/io/prestosql/testng/services/FlakyTestRetryAnalyzer.java:75]: make this warn

others can be debug maybe?

**Review by findepi — APPROVED** [review]

File: presto-testng-services/src/main/java/io/prestosql/testng/services/FlakyTestRetryAnalyzer.java [hunk 2]
```diff
@@ -49,22 +49,30 @@
     @Override
     public boolean retry(ITestResult result)
     {
+        if (result.isSuccess()) {
```
Comment (kokosing) [inline presto-testng-services/src/main/java/io/prestosql/testng/services/FlakyTestRetryAnalyzer.java:52]: separate commit?

File: presto-testng-services/src/main/java/io/prestosql/testng/services/FlakyTestRetryAnalyzer.java (hunk 1 above)
Comment (kokosing) [reply presto-testng-services/src/main/java/io/prestosql/testng/services/FlakyTestRetryAnalyzer.java:75]: Debug are disabled by default turning them on is not trivial.

Comment (kokosing) [inline presto-testng-services/src/main/java/io/prestosql/testng/services/FlakyTestRetryAnalyzer.java:75]: Do not use concatenation with logging

**Review by kokosing — APPROVED** [review]

**Pull request merged**